eclair-core/src/main/scala/fr/acinq/eclair/io/OpenChannelInterceptor.scala

```diff
@@ -0,0 +1,276 @@
+      case nonInitiator: OpenChannelNonInitiator => checkRateLimits(nonInitiator)
+    }
+  }
+
+  private def sanityCheckInitiator(initiator: OpenChannelInitiator): Behavior[Command] = {
+    if (initiator.open.fundingAmount >= Channel.MAX_FUNDING && !initiator.localFeatures.hasFeature(Wumbo)) {
+      initiator.replyTo ! Status.Failure(new RuntimeException(s"fundingAmount=${initiator.open.fundingAmount} is too big, you must enable large channels support in 'eclair.features' to use funding above ${Channel.MAX_FUNDING} (see eclair.conf)"))
+      waitForRequest()
+    } else if (initiator.open.fundingAmount >= Channel.MAX_FUNDING && !initiator.remoteFeatures.hasFeature(Wumbo)) {
+      initiator.replyTo ! Status.Failure(new RuntimeException(s"fundingAmount=${initiator.open.fundingAmount} is too big, the remote peer doesn't support wumbo"))
+      waitForRequest()
+    } else if (initiator.open.fundingAmount > nodeParams.channelConf.maxFundingSatoshis) {
+      initiator.replyTo ! Status.Failure(new RuntimeException(s"fundingAmount=${initiator.open.fundingAmount} is too big for the current settings, increase 'eclair.max-funding-satoshis' (see eclair.conf)"))
+      waitForRequest()
+    } else {
+      // If a channel type was provided, we directly use it instead of computing it based on local and remote features.
+      val channelFlags = initiator.open.channelFlags_opt.getOrElse(nodeParams.channelConf.channelFlags)
+      val channelType = initiator.open.channelType_opt.getOrElse(ChannelTypes.defaultFromFeatures(initiator.localFeatures, initiator.remoteFeatures, channelFlags.announceChannel))
+      val dualFunded = Features.canUseFeature(initiator.localFeatures, initiator.remoteFeatures, Features.DualFunding)
+      val upfrontShutdownScript = Features.canUseFeature(initiator.localFeatures, initiator.remoteFeatures, Features.UpfrontShutdownScript)
+      val localParams = createLocalParams(nodeParams, initiator.localFeatures, upfrontShutdownScript, channelType, isInitiator = true, dualFunded = dualFunded, initiator.open.fundingAmount, initiator.open.disableMaxHtlcValueInFlight)
+      peer ! Peer.SpawnChannelInitiator(initiator.open, ChannelConfig.standard, channelType, localParams, initiator.replyTo.toClassic)
+      waitForRequest()
+    }
+  }
+
+  private def checkRateLimits(nonInitiator: OpenChannelNonInitiator): Behavior[Command] = {
+    val adapter = context.messageAdapter[PendingChannelsRateLimiter.Response](PendingChannelsRateLimiterResponse)
+    pendingChannelsRateLimiter ! AddOrRejectChannel(adapter, nonInitiator.remoteNodeId, nonInitiator.temporaryChannelId)
+    receiveCommandMessage[CheckRateLimitsCommands](context, "checkRateLimits") {
+      case PendingChannelsRateLimiterResponse(PendingChannelsRateLimiter.AcceptOpenChannel) =>
+        validateRemoteChannelType(nonInitiator.temporaryChannelId, nonInitiator.channelFlags, nonInitiator.channelType_opt, nonInitiator.localFeatures, nonInitiator.remoteFeatures) match {
```

Comment:
I think it would make more sense to start with a `sanityCheckNonInitiator()` first to run all _synchronous_ checks, and afterwards run the _asynchronous_ checks (rate-limiting and plugin).

Reply:
Good idea; fixed in [3c4137f](https://github.com/ACINQ/eclair/pull/2552/commits/3c4137fdb0bb6c8f712c40f508c9fc0468c13304).